src/commands/ls.ts

```diff
@@ -74,8 +79,10 @@ const ls = async (
 ) => {
   const authn = await authenticate();
 
-  const isAdminCommand =
-    args.arguments.includes("--all") || args.arguments.includes("--principal");
+  const principal = argValue(args.arguments, "principal");
```

Comment:
why are we not using yargs for this? shouldn't it be in the yargs object? 🤔

Reply:
Quoted args work well already. Added some unit tests to prevent regression.